[fastdds] build failure - file does not exist - fast-discovery-serverd-1.0.1

### Operating system

Ubuntu 22.04.5 LTS in a VM

### Compiler

GCC 11.4.0

### Steps to reproduce the behavior

```Shell
git init
git submodule add https://github.com/microsoft/vcpkg.git
cd vcpkg
./bootstrap-vcpkg.sh
cd ..


Create files:
- `CMakeLists.txt`:
  
  project(test)
  find_package(fastdds CONFIG REQUIRED)
  
- `vcpkg.json`:
  
  {
    "dependencies": [
      "fastdds"
    ]
  }
  


mkdir build
cd build
cmake .. -DCMAKE_TOOLCHAIN_FILE="../vcpkg/scripts/buildsystems/vcpkg.cmake"
```

### Failure logs

```
-- Running vcpkg install - done
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /home/devuser/code/vcpkg_troubleshoot/build/vcpkg_installed/x64-linux/debug/lib/libcrypto.a (found version "3.4.0")  
CMake Error at build/vcpkg_installed/x64-linux/share/fastdds/fast-discovery-server-targets.cmake:78 (message):
  The imported target "fastdds::fast-discovery-server" references the file

     "/home/devuser/code/vcpkg_troubleshoot/build/vcpkg_installed/x64-linux/tools/fastdds/fast-discovery-serverd-1.0.1"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/home/devuser/code/vcpkg_troubleshoot/build/vcpkg_installed/x64-linux/share/fastdds/fast-discovery-server-targets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  build/vcpkg_installed/x64-linux/share/fastdds/fastdds-config.cmake:117 (include)
  vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  CMakeLists.txt:2 (find_package)


CMake Error in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.22)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".


-- Configuring incomplete, errors occurred!
See also "/home/devuser/code/vcpkg_troubleshoot/build/CMakeFiles/CMakeOutput.log".
```

[CMakeOutput.log](https://github.com/user-attachments/files/18682326/CMakeOutput.log)

### Additional context

The issue has been reported before but on the prior port (fastrtps) before rename to fastdds: https://github.com/microsoft/vcpkg/issues/39375. There is a PR here #39482 for that old port however it stalled in July 2024.

I had a look at changes on that PR and it applies also to latest port. Patch attached. Is there a port maintainer who can look at this (e.g. consider what is the best approach) or should I attempt a PR?

[Changes-On-cf035d.patch](https://github.com/user-attachments/files/18682404/Changes-On-cf035d.patch)

--- a/ports/fastdds/portfile.cmake
+++ b/ports/fastdds/portfile.cmake
@@ -71,7 +71,9 @@ elseif(VCPKG_TARGET_IS_LINUX)
 endif()
 
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/fastdds/discovery/parser.py" "tool_path / '../../../bin'" "tool_path / '../../${PORT}'")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/fastdds/fast-discovery-server-targets-debug.cmake" "${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-serverd-1.0.1" "${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-server-1.0.1")
 
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR NOT VCPKG_TARGET_IS_WINDOWS)
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")

--- a/ports/fastdds/portfile.cmake
+++ b/ports/fastdds/portfile.cmake
@@ -71,7 +71,9 @@ elseif(VCPKG_TARGET_IS_LINUX)
 endif()
 
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/fastdds/discovery/parser.py" "tool_path / '../../../bin'" "tool_path / '../../${PORT}'")
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/fastdds/fast-discovery-server-targets-debug.cmake" [[${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-serverd-1.0.1]] [[${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-server-1.0.1]])
+if( VCPKG_BUILD_TYPE STREQUAL "debug")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/fastdds/fast-discovery-server-targets-debug.cmake" [[${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-serverd-1.0.1]] [[${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-server-1.0.1]])
+endif()
 
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR NOT VCPKG_TARGET_IS_WINDOWS)

--- a/ports/fastdds/portfile.cmake
+++ b/ports/fastdds/portfile.cmake
@@ -71,7 +71,7 @@ elseif(VCPKG_TARGET_IS_LINUX)
 endif()
 
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/fastdds/discovery/parser.py" "tool_path / '../../../bin'" "tool_path / '../../${PORT}'")
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/fastdds/fast-discovery-server-targets-debug.cmake" "${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-serverd-1.0.1" "${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-server-1.0.1")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/fastdds/fast-discovery-server-targets-debug.cmake" [[${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-serverd-1.0.1]] [[${_IMPORT_PREFIX}/tools/fastdds/fast-discovery-server-1.0.1]])
 
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR NOT VCPKG_TARGET_IS_WINDOWS)

--- a/ports/fastdds/usage
+++ b/ports/fastdds/usage
@@ -1,0 +1,5 @@
+fastdds provides CMake targets:
+
+  find_package(fastdds CONFIG REQUIRED)
+  target_link_libraries(main PRIVATE fastdds)
+    

--- a/ports/fastdds/vcpkg.json
+++ b/ports/fastdds/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "fastdds",
   "version": "3.1.2",
+  "port-version": 1,
   "description": "eprosima Fast DDS (formerly Fast RTPS) is a C++ implementation of the DDS (Data Distribution Service) standard of the OMG (Object Management Group). eProsima Fast DDS implements the RTPS (Real Time Publish Subscribe) protocol, which provides publisher-subscriber communications over unreliable transports such as UDP, as defined and maintained by the Object Management Group (OMG) consortium.",
   "homepage": "https://www.eprosima.com/",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2694,7 +2694,7 @@
     },
     "fastdds": {
       "baseline": "3.1.2",
-      "port-version": 0
+      "port-version": 1
     },
     "fastfeat": {
       "baseline": "391d5e9",

--- a/versions/f-/fastdds.json
+++ b/versions/f-/fastdds.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ec15df54e51a1b0f9c858d0c8e2fb4769820845f",
+      "git-tree": "ddee7c4609b3ee74bca1d95003217c7847b39fc4",
       "version": "3.1.2",
       "port-version": 1
     },

--- a/versions/f-/fastdds.json
+++ b/versions/f-/fastdds.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "2cfcd7a0a42583ee59b0a588dc88b3502099cd4a",
+      "git-tree": "ec15df54e51a1b0f9c858d0c8e2fb4769820845f",
       "version": "3.1.2",
       "port-version": 1
     },

--- a/versions/f-/fastdds.json
+++ b/versions/f-/fastdds.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2cfcd7a0a42583ee59b0a588dc88b3502099cd4a",
+      "version": "3.1.2",
+      "port-version": 1
+    },
+    {
       "git-tree": "583c25bed0e0994ddb7c8b647f69508684a5f0b4",
       "version": "3.1.2",
       "port-version": 0

